src/goto-synthesizer/cegis_verifier.cpp

```diff
@@ -211,6 +211,62 @@ std::list<loop_idt> cegis_verifiert::get_cause_loop_id(
   return result;
 }
 
+cext::violation_locationt cegis_verifiert::get_violation_location(
+  const loop_idt &loop_id,
+  const goto_functiont &function,
+  unsigned location_number_of_target)
+{
+  if(is_instruction_in_transfomed_loop_condition(
+       loop_id, function, location_number_of_target))
+  {
+    return cext::violation_locationt::in_condition;
+  }
+
+  if(is_instruction_in_transfomed_loop(
+       loop_id, function, location_number_of_target))
+  {
+    return cext::violation_locationt::in_loop;
+  }
+
+  return cext::violation_locationt::after_loop;
+}
+
+bool cegis_verifiert::is_instruction_in_transfomed_loop_condition(
+  const loop_idt &loop_id,
+  const goto_functiont &function,
+  unsigned location_number_of_target)
+{
+  // The transformed loop condition are instructions from
+  // loop havocing instructions
+  // to
+  // if(!guard) GOTO EXIT
+  unsigned location_number_of_havocing = 0;
+  for(goto_programt::const_targett it = function.body.instructions.begin();
+      it != function.body.instructions.end();
+      ++it)
+  {
```

Comment:
```suggestion
  for(auto it = function.body.instructions.begin();
      it != function.body.instructions.end();
      ++it)
  {
    unsigned location_number_of_havocing = 0;
```

Reply:
`location_number_of_havocing` is assigned and used in different loop iterations. It should not be a loop-local here.